File: answers/leetcode/Longest Increasing Subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        std::vector<int> dp(nums.size(), 1);
        int length = nums.size() > 0;
        for (size_t i = 0; i < dp.size(); ++ i) {
            for (size_t j = 0; j < i; ++ j) {
                if (nums[j] < nums[i]) {
                    dp[i] = max(dp[i], dp[j] + 1);
                    length = max(length, dp[i]);
                }
            }
        }
        return length;
    }
};

File: answers/leetcode/Longest Increasing Subsequence/Longest Increasing Subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int maxLength = 0;
        int[] lengths = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            lengths[i] = 1;
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) {
                    lengths[i] = Math.max(lengths[i], lengths[j] + 1);
                }
            }
            maxLength = Math.max(maxLength, lengths[i]);
        }
        return maxLength;
    }
}